components/properties/src/props.rs

```diff
@@ -428,6 +570,36 @@ impl_value_getter! {
         /// assert_eq!(lookup.get_loose("Animated_Gif"), None);
         /// ```
         pub fn get_name_to_enum_mapper();
+        /// Return a [`PropertyEnumToValueNameMapper`], capable of looking up short names
+        /// for values of the `General_Category` enumerated property
+        ///
+        /// # Example
+        ///
+        /// ```
+        /// use icu::properties::GeneralCategory;
+        ///
+        /// let lookup = GeneralCategory::get_enum_to_short_name_mapper(&icu_testdata::unstable())
+        ///                  .expect("The data should be valid");
+        /// let lookup = lookup.as_borrowed();
+        /// assert_eq!(lookup.get(GeneralCategory::UppercaseLetter), Some("Lu"));
+        /// assert_eq!(lookup.get(GeneralCategory::DashPunctuation), Some("Pd"));
+        /// ```
+        pub fn get_enum_to_short_name_mapper();
```

Comment:
q: should GeneralCategory have these? Data is not needed to do this for GC, we could easily have special methods that just call a match. We could keep the mappers for symmetry.

Reply:
Yes? Seems like properties with mapping in one direction should have it in the other. Also for GCG. I don't like special casing just GC and GCG.

Reply:
^ Are you doing GCG in this PR?

Reply:
@sffc No, you misunderstand my question about GC: should I do _mappers_ for GC? I could instead just have `.short_name()` and `.long_name()`.

For short and long names those can never change; we do not _need_ data-driven mappers, we can have it be code-driven, but it would be a separate codepath.


I was punting GCG because it's also not clear if that should be data-driven: yes, new mask values could be added in the future, but the TrieValue conversion _already_ hardcodes mask values (it needs to be data driven because we might add additional aliases but it can't handle new long/short names). To me it feels like there is no benefit from that being data driven.

Reply:
Right, that's what I meant by "I don't like special casing just GC and GCG"

Reply:
Yeah I can file a followup about this.

I would like to have the code-driven API though, even if we support a data-driven API we discourage.